internal/cli/analytics/reporter.go

```diff
@@ -0,0 +1,29 @@
+package analytics
+
+const (
+	defaultCliVersion = "v9.99.9-dev"
+)
+
+var (
+	// APIKey contains the API key for external analytics service. It is set during application build.
+	APIKey string = ""
+)
+
+// Reporter defines behavior for reporting analytics.
+type Reporter interface {
+	ReportCommand(cmd string) error
+	ReportError(err error, cmd string) error
+	Close()
+}
+
+// NewReporter creates a new Reporter instance.
+func NewReporter() Reporter {
```

Comment:
I think it would be better to return here noop reporter in case the analytics are disabled. This would be consistent without approach with Botkube Agent. In that way we won't report anything even if someone by accident executes a `Report` method in the code. WDYT?

Also, I'd suggest renaming this to `GetReporter()` or something like this as this is a conditional constructor and it's not clear from the usage perspective.